### ISSUE TITLE
`onMapLoaded` in iOS with Google Maps fires multiple times

### ISSUE THREAD

**Issue by n-ii-ma** [react-native-maps/react-native-maps#4566]
### Summary

I'm fitting my supplied markers to the screen when the map has finished its loading by using `onMapLoaded`. This props works as expected on Android but on iOS with Google Maps enabled, `onMapLoaded` gets called multiple times when moving the map, causing all kinds of unexpected behavior.

Using `onMapReady` doesn't cause this issue but even that has its own issues and limitations since it doesn't seem to track the loading of the map at all.

### Reproducible sample code

```TSX
// Animate map view so that markers are in the view
  const handleMapReady = useCallback(() => {
    // Set map loading to false
    setIsLoading(false);

    //////////////////////////////////////
    console.log('GETTING CALLED');
    //////////////////////////////////////

    // Extract markers id
    const markersId = markers.map(marker => marker.id);

    // Set timeout for performance
    const waiting = setTimeout(() => {
      mapRef.current?.fitToSuppliedMarkers(markersId, {
        edgePadding: {
          top: Platform.OS === 'android' ? 100 : 50,
          bottom: Platform.OS === 'android' ? 0 : 25,
          left: 25,
          right: 25,
        },
        animated: true,
      });
    }, 1000);

    return () => clearTimeout(waiting);
  }, [markers]);

....................................................

<MapView
  provider="google"
  ref={mapRef}
  onMapLoaded={handleMapReady}
  loadingEnabled>
  {markers.map(marker => (
    <Marker
       identifier={marker.id}
       title={marker.title}
       coordinate={marker.coordinate}
       key={marker.id}
       image={marker.path}
     />
  ))}
</MapView>
```

_The supplied markers which are in the dependency array of `useCallback()` never change and aren't dynamic._

### Steps to reproduce

Make sure that Google Maps is enabled on iOS and set up a function to be called after the map is loaded and pass it to `onMapLoaded`.

Even something as simple as this would be enough:
```
...
onMapLoaded={() => console.log('GETTING CALLED')}
...
```

### Expected result

I expect the function of `handleMapReady` to be called only once after the map has loaded, so that `GETTING CALLED` is logged only once.

### Actual result

The function of `handleMapReady` is getting called multiple times so I'm not able to move the map at all since the `fitToSuppliedMarkers` method is being invoked multiple times (`GETTING CALLED` is also getting logged multiple times).

### React Native Maps Version

1.3.2

### What platforms are you seeing the problem on?

iOS (Google Maps)

### React Native Version

0.70.6

### What version of Expo are you using?

Not using Expo

### Device(s)

iPhone 14

**Comment by AliMSbaee** [comment]
Same issue

**Comment by polygbrl** [comment]
I think you might be experiencing [what I described in an other issue](https://github.com/react-native-maps/react-native-maps/issues/4244#issuecomment-1410358608). You should assign the map ref to yours like so `ref={map => mapRef = map}` and not directly `ref={mapRef}`.

**Comment by anas-araid** [comment]
even with this solution I still have the same issue, is there any other workaround we can try?

**Comment by Simon-TechForm** [comment]
Hello 👋

`onMapLoaded` is actually expected to fire multiple times (e.g. whenever new tiles are rendered). I've just released a new version that includes documentation regarding how `onMapLoaded` - now `onTilesRendered` is expected to behave.

@Peeeep should be onto something in regards to using `onMapReady` instead. What's the use case that prevents using `onMapReady`?

I hope you are okay with me closing this for, since this is the expected behaviour for `onMapLoaded`. Feel free to open a new issue if you experience any bugs in the newest version or alternatively start a new discussion regarding how to satisfy your use case.
